src/System.Net.Security/tests/FunctionalTests/SslStreamStreamToStreamTest.cs

```diff
@@ -28,9 +29,16 @@ public void SslStream_StreamToStream_Authentication_Success()
                 Task[] auth = new Task[2];
                 auth[0] = client.AuthenticateAsClientAsync(certificate.Subject);
                 auth[1] = server.AuthenticateAsServerAsync(certificate);
-
+                
                 bool finished = Task.WaitAll(auth, TimeSpan.FromSeconds(3));
                 Assert.True(finished, "Handshake completed in the allotted time");
+
+                string testData = "Hello SSL";
+                byte[] sendBuf = Encoding.UTF8.GetBytes(testData);
+                client.Write(sendBuf);
+                byte[] recvBuf = new byte[sendBuf.Length];
+                server.Read(recvBuf, 0, recvBuf.Length);
```

Comment:
In general, Stream.Read does not guarantee that it reads as much data as you request.  I don't know if SslStream layers additional guarantees on top of that, but you might need a loop here.